### PR TITLE
Using the user_context to not pass around the token info.

### PR DESCRIPTION
**Pull request by sumesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/7434

*Description of changes:*
- Use user_context for injecting token information

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/models/user.py [hunk 1]
```diff
@@ -111,30 +118,32 @@ def create_from_jwt_token(cls, token: dict, first_name: str, last_name: str):
...
+        user_from_context: UserContext = kwargs['user_context']
+        token = user_from_context.token_info
+        if not token or not user:
             return None
 
         # Do not save if nothing has been changed
         # pylint: disable=too-many-boolean-expressions
         if not is_login \
-                and user.username == token.get('preferred_username', user.username) \
+                and (user.username == user_from_context.user_name or user.username) \
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/models/user.py:133]: the preferred_username is not needed? in the PR , user_name is used

File: auth-api/src/auth_api/services/user.py [hunk 2]
```diff
@@ -309,25 +312,27 @@ def _create_kc_user(membership):
...
         request_json = {} if not request_json else request_json
 
-        is_anonymous_user = token.get('accessType', None) == AccessType.ANONYMOUS.value
+        is_anonymous_user = user_from_context.token_info.get('accessType', None) == AccessType.ANONYMOUS.value
         if not is_anonymous_user:
-            existing_user = UserModel.find_by_jwt_token(token)
+            existing_user = UserModel.find_by_jwt_token()
         else:
-            existing_user = UserModel.find_by_username(token.get('preferred_username'))
+            existing_user = UserModel.find_by_username(user_from_context.user_name)
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/services/user.py:328]: same here..user_name instead of prefered

File: auth-api/src/auth_api/utils/user_context.py [hunk 3]
```diff
@@ -33,23 +33,14 @@ class UserContext:  # pylint: disable=too-many-instance-attributes
...
-        user_model: UserModel = None
-        if token_info:
-            user_model = UserModel.find_by_jwt_token(token_info)
-
-        if user_model:
-            self._user_id = user_model.id
-        # try to user name from DB ;if not avaiable fall back to token_info
-        self._user_name: str = getattr(user_model, 'username',
-                                       token_info.get('username', token_info.get('preferred_username', None)))
+        self._user_name: str = token_info.get('username', token_info.get('preferred_username', None))
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/utils/user_context.py:38]: I think now i understand why u went for username ;so can ignore above pr comments

File: auth-api/src/auth_api/utils/user_context.py [hunk 4]
```diff
@@ -119,14 +105,27 @@ def token_info(self) -> Dict:
         """Return the name."""
         return self._token_info
 
+    @property
+    def account_id(self) -> Dict:
+        """Return the account id."""
+        account_id = _get_token_info().get('Account-Id', None)
```
Comment (saravanpa-aot) [inline auth-api/src/auth_api/utils/user_context.py:111]: the only time we have account id is the authorization request...
hope nobody will use this method in other cases where we need an org id

**Review by saravanpa-aot — APPROVED** [review]

File: auth-api/src/auth_api/utils/user_context.py (hunk 4 above)
Comment (sumesh-aot) [reply auth-api/src/auth_api/utils/user_context.py:111]: Yea, I think in most of the other cases we have org_id in the payload or in path.

**Pull request merged**